### PR TITLE
Stop pinning the versions of the cookbooks to absurdly old versions

### PR DESCRIPTION
**Pull request by michaelb990**
With https://github.com/socrata/aws/pull/44, we can specify cookbook specific Berksfiles to use for building AMIs. Jenkins workers don't run chef once they are launched and have nothing stopping them from being built with the latest and greatest community cookbooks. 

This changes the docker service to use `overlay` as the storage driver. We have been having some issues with devicemapper.

It also removes some code that was dropping SBT configuration that we would prefer to drop at instance launch.

### CODE REVIEW

File: recipes/_docker.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
-include_recipe 'docker'
+docker_service 'default' do
+  storage_driver 'overlay'
```
Comment (jhaynes) [inline recipes/_docker.rb]: this might want to be `aufs` to match what we do in mesos unless you've got a good reason not to.

File: Gemfile.lock [hunk 2]
```diff
@@ -1,134 +0,0 @@
-GEM
```
Comment (jhaynes) [inline Gemfile.lock]: @RoboticCheese Do you have strong feelings one way or the other about Berksfile.lock and Gemfile.lock?

File: recipes/_sbt.rb [hunk 3]
```diff
@@ -24,25 +24,6 @@
   mode 0o755
 end
 
-if Chef::Config.solo
-  jenkins_credentials = data_bag_item('credentials', 'jenkins')
-else
-  jenkins_credentials = Chef::EncryptedDataBagItem.load('credentials', 'jenkins')
-end
-
-if artifactory = jenkins_credentials['artifactory'][node.environment]
```
Comment (jhaynes) [inline recipes/_sbt.rb]: How will they get the artifactory creds now?

File: recipes/_docker.rb (hunk 1 above)
Comment (michaelb990) [inline recipes/_docker.rb]: I wanted to try the new stuff 😢

File: recipes/_sbt.rb (hunk 3 above)
Comment (michaelb990) [inline recipes/_sbt.rb]: They never got them this way to begin with... We populate it with clortho when the worker is launched by jenkins.

File: Gemfile.lock (hunk 2 above)
Comment (michaelb990) [inline Gemfile.lock]: Let's fight about this!

**Pull request merged**